### PR TITLE
Added check command generator

### PR DESCRIPTION
**Pull request by mawaishanif**
Added feature #28 with doc and tests.

### CODE REVIEW

File: src/Commands/HealthCheckMakeCommand.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * @var string
+     */
+    protected $signature = 'make:check {name : The name of the check class}';
+
+    /**
+     * The check command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create a new check class';
```
Comment (tylerwoonton) [inline src/Commands/HealthCheckMakeCommand.php:21]:
```suggestion
    protected $description = 'Create a new health check class';
```

File: stubs/check.stub [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    public function __construct()
+    {
+        //
+    }
+
+    public function status()
+    {
+        return $this->okay();
+    }
+}
```
Comment (Gman98ish) [inline stubs/check.stub:21]: Would be good to have a new line here, it'd stop phpcs from flagging it

Comment (tylerwoonton) [inline stubs/check.stub:21]:
```suggestion
}

```

File: src/Commands/HealthCheckMakeCommand.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+use Illuminate\Console\GeneratorCommand;
+
+class HealthCheckMakeCommand extends GeneratorCommand
+{
+    /**
+     * The health check name.
+     *
+     * @var string
+     */
+    protected $signature = 'make:check {name : The name of the check class}';
```
Comment (tylerwoonton) [inline src/Commands/HealthCheckMakeCommand.php:14]:
```suggestion
    protected $signature = 'make:check {name : The name of the health check class}';
```

**Review by Gman98ish — CHANGES_REQUESTED** [review]
This looks really good, very small change to do with the stub, but the code itself looks great.

The 7.1 tests are failing, I'm not entirely sure why. I'll investigate a bit later

**Review by tylerwoonton — CHANGES_REQUESTED** [review]
Thanks for submitting a PR @mawaishanif 

I've highlighted a couple of quick changes which you can just apply.

File: stubs/check.stub (hunk 2 above)
Comment (tylerwoonton) [reply stubs/check.stub:21]: I'll resolve this as I've added a suggestion for it below. 

https://github.com/ukfast/laravel-health-check/pull/30#discussion_r498067685

File: tests/Commands/HealthCheckMakeCommandTest.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        return ['UKFast\HealthCheck\HealthCheckServiceProvider'];
+    }
+
+    /**
+     * @test
+     */
+    public function creates_a_new_check()
+    {
+        $checkName = "TestCheck";
+        $checkClassFile = $this->app->basePath("app/Checks/{$checkName}.php");
```
Comment (Gman98ish) [inline tests/Commands/HealthCheckMakeCommandTest.php:21]: This is the other thing I'm not sure about, would we be able to put this in a tests/stubs folder which we commit to `.gitignore`

If the test fails before it gets a chance to unlink it, we could end up with these test stubs accidentally comitted

Comment (Gman98ish) [reply tests/Commands/HealthCheckMakeCommandTest.php:21]: In fact, ignore this. Not like there are any checks in the app folder of this project

**Comment by mawaishanif** [comment]
Thanks @Gman98ish  @tylerwoonton

File: tests/Commands/HealthCheckMakeCommandTest.php (hunk 4 above)
Comment (mawaishanif) [reply tests/Commands/HealthCheckMakeCommandTest.php:21]: What we can do is to check for the file and unlink it before creating and asserting it.

**Comment by miff2000** [comment]
This needs rebasing onto `master`. I had a go but there were merge conflicts with it. Any chance you could have a look @tylerwoonton?

**Comment by tylerwoonton** [comment]
> This needs rebasing onto `master`. I had a go but there were merge conflicts with it. Any chance you could have a look @tylerwoonton?

Sure, I'll take a look @miff2000.

**Pull request closed without merging**